lib/logstash/codecs/es_bulk.rb

```diff
@@ -27,17 +49,17 @@ def decode(data)
         line = LogStash::Json.load(bulk.get("message"))
         case @state
         when :metadata
-          event = LogStash::Event.new(line)
-          event.set("@metadata", @metadata)
+          event = targeted_event_factory.new_event(line)
+          event.set(@metadata_field, @metadata)
           yield event
```

Comment:
would it make sense to `event.set('[event][original]', bulk.get("message").dup.freeze)` in ECS mode?

Reply:
There are four options here. Considering the following input
```
      { "index" : { "_index" : "test", "_type" : "type1", "_id" : "1" } }
      { "field1" : "value1" }
      { "delete" : { "_index" : "test", "_type" : "type1", "_id" : "2" } }
      { "create" : { "_index" : "test", "_type" : "type1", "_id" : "3" } }
      { "field1" : "value3" }
      { "update" : {"_id" : "1", "_type" : "type1", "_index" : "index1"} }
      { "doc" : {"field2" : "value2"} }
```

1. Store the whole input as `event.original`.
The input generates four events. Each event saves the complete raw data.
It could generate too much extra bandwidth

2. Store { "field1" : "value1" } as `event.original`.
This is the line when target_event_factory is called. Metadata is not included. The information is not complete. 
This is the result of `event.set('[event][original]', bulk.get("message").dup.freeze)`
The information is not particularly interesting because the same stuff transforms into an event. 

3. tailor-make a new line to concatenate metadata  { "index" : { "_index" : "test", "_type" : "type1", "_id" : "1" } } and data  { "field1" : "value1" }
This will be a full event that requires concatenating the decoded string

4. do not store `event.original`

I am opened to save it in `event.original` if you think it is useful

Reply:
oh right, it's tricky with the multi-line and the managed state ... was undecided on this making sense.
Reading your comments I am leaning towards **4.** do not store `event.original` for now and ship as is...